businessRules/users.js

```diff
@@ -4,61 +4,72 @@ const { createUser } = require('../models/users');
 const { getUsers, setUsers } = require('../utils/storage');
 
 function registerUser(userName, password, firstName, lastName) {
+    if (!userName) {
+        throw new Error('Error. There was no username provided.');
+    }
+    if (!password) {
+        throw new Error('Error. There was no password provided.');
+    }
+
     firstName = firstName || 'Anonymous';
     lastName = lastName || 'Incognito';
 
     createUser(userName, password, firstName, lastName);
 }
 
-function login(req, res) {
-    let users = getUsers();
+function login(userName, password) {
+    if (!userName) {
+        throw new Error('Error. There was no username provided.');
+    }
+    if (!password) {
+        throw new Error('Error. There was no password provided.');
+    }
 
-    const userName = req.body.userName;
-    const password = md5(req.body.password);
+    let users = getUsers();
 
     const user = users.find(
         (user) => user.userName === userName && user.password === password
     );
 
     if (!user) {
-        res.status(400).json({ status: 'No such user.' });
-        return;
+        throw new Error('Error. No such user.');
     }
 
     const sessionId = md5(v4()); //Should be REAL cryptography.
     user.session = sessionId;
-    setUsers(users);
-    res.cookie('userLoginSession', sessionId, {
-        sameSite: 'None',
-        secure: true,
-        httpOnly: true,
-        maxAge: 900000,
-    });
-    res.status(200).json({
-        status: 'OK',
-        username: user.userName,
-        statusMessage: 'You have successfully logged in.',
-    });
+    setUsers(user);
+
+    return sessionId;
 }
 
 function usersList() {
     return getUsers();
 }
 
-function logout(req, res) {
-    res.clearCookie('userLoginSession', {
-        sameSite: 'None',
-        secure: true,
-        httpOnly: true,
-        sameSite: 'None',
-    });
-    res.status(200).json({
-        status: 'OK',
-        statusMessage: 'You have successfully logged out.',
-    });
+function logout(userLoginSession) {
+    if (!userLoginSession) {
+        throw new Error('Error. There was no users session data provided.');
+    }
+
+    let users = getUsers();
+
+    const user = userLoginSession
+        ? users.find((user) => user.session === userLoginSession)
+        : null;
+    if (!user) {
+        throw new Error('You are not logged in.');
+    }
+    updateUser(user.id, 'session', 'null');
```

Comment:
Why `null` is a string here?

Reply:
If I'm sending null = than my if (!value) state works, and throws error.

Reply:
Calling `updateUser` function and passing `null` as a string is a sign of a poor function design - even if it works.

Try to refactor `updateUser` function so that you could call it like this:

```js
updateUser(user.id, 'session', null);
```

A more readable function call would look like this:

```js
deleteUserSession(user.id);
```

Reply:
updateUser fixed, commited.
about deleteUser, wouldn't it be a bit confusing, when in find method it would be written user.id = user.id?
Is it a good practice? Just curios.

Reply:
updateUser fixed and commited.
And you're advice is to rename the function?

Reply:
Not to rename, but to create a new function that uses `updateUser`. For example:
```js
function deleteUserSession(userId) {
  updateUser(userId, 'session', null);
}
```

You can then use `deleteUserSession` in `logout`.

Reply:
Oooooh, devide and conquer. functionality should be devided. Right. 
Updated, commited.